### PR TITLE
Move rss feed to controller

### PR DESCRIPTION
**Pull request by nickvergessen**

### CODE REVIEW

**Comment by nickvergessen** [comment]
@LukasReschke

File: controller/feed.php [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		if ($this->request->getHeader('accept') !== null && stristr($this->request->getHeader('accept'), 'application/rss+xml')) {
+			$response->addHeader('Content-Type', 'application/rss+xml');
+		} else {
+			$response->addHeader('Content-Type', 'text/xml; charset=UTF-8');
+		}
+
+		return $response;
+	}
+
+	protected function getUser() {
```
Comment (LukasReschke) [inline controller/feed.php]: PHPDoc

File: controller/feed.php [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		if (!$this->session->isLoggedIn()) {
+			return $this->getUserFromToken();
+		} else {
+			return $this->session->getUser()->getUID();
+		}
+	}
+
+	protected function getUserFromToken() {
```
Comment (LukasReschke) [inline controller/feed.php]: PHPDoc

File: lib/data.php [hunk 3]
```diff
@@ -138,31 +138,22 @@ public static function storeMail($app, $subject, array $subjectParams, $affected
...
-	}
-
-	/**
 	 * @brief Read a list of events from the activity stream
 	 * @param GroupHelper $groupHelper Allows activities to be grouped
 	 * @param UserSettings $userSettings Gets the settings of the user
 	 * @param int $start The start entry
 	 * @param int $count The number of statements to read
 	 * @param string $filter Filter the activities
+	 * @param string $user User for whome we display the stream
```
Comment (LukasReschke) [inline lib/data.php]: whome?

File: lib/data.php [hunk 4]
```diff
@@ -138,31 +138,22 @@ public static function storeMail($app, $subject, array $subjectParams, $affected
...
 	 * @return array
 	 */
-	public function read(GroupHelper $groupHelper, UserSettings $userSettings, $start, $count, $filter = 'all') {
+	public function read(GroupHelper $groupHelper, UserSettings $userSettings, $start, $count, $filter = 'all', $user = '') {
 		// get current user
-		$user = User::getUser();
+		$user = ($user !== '') ? $user : User::getUser();
```
Comment (LukasReschke) [inline lib/data.php]: Yay - static code :see_no_evil:

File: lib/display.php [hunk 5]
```diff
@@ -33,18 +33,44 @@
...
+
+	/**
+	 * Constructor
+	 *
+	 * @param \OCP\IDateTimeFormatter $dateTimeFormatter
+	 * @param \OCP\IPreview $preview
+	 * @param \OCP\IURLGenerator $urlGenerator
+	 * @param View $view
+	 */
+	public function __construct(\OCP\IDateTimeFormatter $dateTimeFormatter, \OCP\IPreview $preview, \OCP\IURLGenerator $urlGenerator, \OC\Files\View $view) {
```
Comment (LukasReschke) [inline lib/display.php]: Newline for every entry? :smile:

File: templates/stream.list.php [hunk 6]
```diff
@@ -52,7 +54,7 @@
 	<div class="boxcontainer">
 <?php
 	}
-	echo \OCA\Activity\Display::show($event);
+	echo $displayHelper->show($event);
```
Comment (LukasReschke) [inline templates/stream.list.php]: `print_unescaped`? (just to have the same usage everywhere in core apps)

**Comment by LukasReschke** [comment]
Great stuff :+1:

**Comment by nickvergessen** [comment]
Fixed all the comments

**Pull request merged**